src/app/staticanalysis/modules/abstractAstView.js

```diff
@@ -0,0 +1,93 @@
+var common = require('./staticAnalysisCommon')
+var AstWalker = require('ethereum-remix').util.AstWalker
+
+function abstractAstView () {
+  this.contracts = null
+  this.currentContractIndex = null
+  this.currentFunctionIndex = null
+  this.currentModifierIndex = null
+  this.isFunctionNotModifier = false
+}
+
+abstractAstView.prototype.builder = function (relevantNodeFilter, contractsOut) {
```

Comment:
OK i will comment a bit ;) , Yes those are the nodes that are relevant for your analysis. For example if you are only interessted in variable declarations within functions you can filter here to make the analysis code easier